### PR TITLE
Do not require cython for normal builds

### PR DESCRIPTION
**Pull request by opoplawski**
## Description

We can build without cython when using pypi source tarballs.  Let people do that.
## Checklist

[It's fine to submit PRs which are a work in progress! But before they are merged, all PRs should provide:]
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Unit tests

[For detailed information on these and other aspects see [scikit-image contribution guidelines](http://scikit-image.org/docs/dev/contribute.html)]
## References

Closes issue #2035

### CODE REVIEW

File: skimage/_build.py [hunk 1]
```diff
@@ -27,15 +27,19 @@ def cython(pyx_files, working_path=''):
     try:
         from Cython import __version__
         if LooseVersion(__version__) < '0.23':
-            raise RuntimeError('Cython >= 0.23 needed to build scikit-image')
+            raise ImportError
```
Comment (stefanv) [inline skimage/_build.py]: Why replace the error with a less informative version?

File: skimage/_build.py [hunk 2]
```diff
@@ -27,15 +27,19 @@ def cython(pyx_files, working_path=''):
...
-        # the distributed .c files
-        print("Cython not found; falling back to pre-built %s" \
-              % " ".join([f.replace('.pyx.in', 'c').replace('.pyx', '.c')
-                          for f in pyx_files]))
+        # If cython is not found, the build will make use of
+        # the distributed .c files if present
+        c_files = [f.replace('.pyx.in', '.c').replace('.pyx', '.c') for f in pyx_files]
+        for cfile in [os.path.join(working_path, f) for f in c_files]:
+            if not os.path.isfile(cfile):
+                raise RuntimeError('Cython >= 0.23 is required to build scikit-image from SCM checkout')
```
Comment (stefanv) [inline skimage/_build.py]: SCM -> git checkout or something more readable

File: skimage/_build.py (hunk 1 above)
Comment (opoplawski) [inline skimage/_build.py]: It will fall down to the later error if needed.

File: skimage/_build.py (hunk 2 above)
Comment (opoplawski) [inline skimage/_build.py]: I was trying to be software agnostic, but I could change it to git if you'd like.

Comment (stefanv) [inline skimage/_build.py]: Oh, that's fine, I don't mind the exact term used, it's more that a lot of users may have no idea what "SCM" means.

**Comment by opoplawski** [comment]
Ping.  Is there anything that I need to do here?

File: skimage/_build.py (hunk 1 above)
Comment (sciunto) [inline skimage/_build.py]: I agree with @stefanv You can raise two different messages. Needs a little refactoring.

Comment (soupault) [inline skimage/_build.py]: :+1: from me to revert this change as well.

File: skimage/_build.py [hunk 3]
```diff
@@ -27,15 +27,19 @@ def cython(pyx_files, working_path=''):
...
-        # the distributed .c files
-        print("Cython not found; falling back to pre-built %s" \
-              % " ".join([f.replace('.pyx.in', 'c').replace('.pyx', '.c')
-                          for f in pyx_files]))
+        # If cython is not found, the build will make use of
+        # the distributed .c files if present
+        c_files = [f.replace('.pyx.in', '.c').replace('.pyx', '.c') for f in pyx_files]
+        for cfile in [os.path.join(working_path, f) for f in c_files]:
+            if not os.path.isfile(cfile):
+                raise RuntimeError('Cython >= 0.23 is required to build scikit-image from git checkout')
```
Comment (soupault) [inline skimage/_build.py]: Could you, please, add a global variable `CYTHON_VERSION` and make use of it? This will be much easier to maintain.

**Comment by soupault** [comment]
Thanks @opoplawski ! Fixed via #2509.
